### ISSUE TITLE
Call for new maintainer(s)

### ISSUE THREAD

**Issue by deepsweet** [svg/svgo#1320]
Hi.

My name is Kir, I'm the original author (2012-2016) of SVGO. Current maintainer @GreLI seems to be quite away from GitHub recently and I think it's time to call for new people interested in the project.

With 10M of weekly downloads from NPM I'm not exactly sure how to choose someone and grant access, it sounds like a great responsibility. I'd probably start with reviewing some PRs from a candidate and then maybe a video call. Just want to make sure it's in good hands.

Anyone?

**Comment by paulocoghi** [comment]
@strarsis is a strong candidate :)

**Comment by strarsis** [comment]
Yes, I would like to merge some of my and other PRs so common issues and feature requests can be resolved.

**Comment by TrySound** [comment]
Hi @deepsweet! I would like to help with refreshing the package and fixing some issues. I already submitted a few pull requests. https://github.com/svg/svgo/pull/1293 https://github.com/svg/svgo/pull/1292 https://github.com/svg/svgo/pull/1291 https://github.com/svg/svgo/pull/1290

**Comment by paulocoghi** [comment]
@strarsis I just don't understand the `mysql` term on your phrase. :smile: I'm a little confused xD

**Comment by sk-** [comment]
In the short term I think something along the following lines should be done, so that users know that the project is indeed not dead and is going to be actively maintained.
- Communicate the plan
- Closing old issues and PRs. I'd close everything older than 2 years. For the rest I'd manually go through it triaging it.
- Create templates for creating issues. For bug reports, require debug information, like version number, options, etc. For questions, redirect users to StackOverflow. For feature requests, ask them to review the new rules requirements (see below) document.
- Review "recent" bug fixes PR pending review, there are  a couple of these authored by some of the people who have shown interest on taking over. This could be a good first filter to granting access to interested developers.
- Define some guidelines on what kind of new rules will be accepted, something akin to what ESlint has for their rules. See https://eslint.org/docs/developer-guide/contributing/new-rules#core-rule-guidelines

For the medium term:
- Modernize codebase (Remove support for EOL node versions, maybe migrate to typescript which can be incremental, etc)
- Improve test suite. I worked on downloading a large SVG dataset from Wikicommons, running `svgo` with defaults options and comparing the rendering of input and output svgs. With that technique I manage to find a bunch of errors.
- Improve documentation

Longer term:
- Audit existing rules. Each rule has its own maintenance cost and probably not all of them are as useful or play nice with the overall project structure.

**Comment by lazee** [comment]
@deepsweet Thank you for this initiative and for your magnificent work with svgo. 

I understand your need to find a trustworthy and good maintainer for this project. Whether that is @strarsis, @TrySound, @sk- (or anybody else) is yet to be seen. Personally, I don't have the time, but I will help with PRs where I can as soon as I know there is a fair chance that it will end up in the codebase.

I agree with all the @sk wrote. I would like to add that it should be kept backward compatible (as there are so many packages that depends on it) and...it would be so cool if the codebase could be split up so that we, somewhere down the road can have Deno support as well ;)

**Comment by TrySound** [comment]
Hi guys. I'm maintainer now. Feel free to ping me in issues or pull requests.

**Comment by strarsis** [comment]
@TrySound: Merging these PRs would be nice:
- https://github.com/svg/svgo/pull/1208 (fix tests)
- https://github.com/svg/svgo/pull/1301 (replace unmaintained sax parser with txml parser)
- https://github.com/svg/svgo/pull/1177 (add missing tests for multipass option)
- https://github.com/svg/svgo/pull/1278 (mergeStyles plugin)
- https://github.com/svg/svgo/pull/1279 (dereferenceUses plugin)

**Comment by yisibl** [comment]
TrySound is an excellent maintainer!

**Comment by yoriiis** [comment]
I'm not sure but the topic still seems relevant to me.

The project seems a bit slowed down, and the number of issues is quite high, ~230 currently. Moreover, there are regressions since the last release and no response on Discord and GitHub.

The project is great, really! But, as a user and contributor, I think it's important to be transparent. Is there a need for support on the project?

The subject is sensitive and is in no way negative, I am also a project maintainer and I can understand that we also have other obligations.

**Comment by deepsweet** [comment]
Hi @yoriiis 

Here is my comment regarding this from another thread https://github.com/svg/svgo/issues/1128#issuecomment-1405767522:

<blockquote>
I'm the original author as it says in the `package.json`. I have all the possible access both here and on NPM, but not the time and necessary context to continue working on the project. For many years.

At the same time I can't just blindly merge PRs to the project with 15M NPM downloads per week, so I've tried to look for maintainers over the years. It's way more complicated than it seems to be, starting from the observation that there are no many candidates. In fact, there are none. One thing is to merge old and stalled for months PRs, and another is to develop it all further and push forward.

The only realistic way to dramatically change this situation is to contribute and be ready to become a new maintainer.

Talk to me.
</blockquote>

Re-opening and pinning this issue again.

**Comment by GreLI** [comment]
It never works that way, unfortunately. More realistic is getting some big sponsorship from large companies.

Also, project like this requires at least a couple of full-time maintainers to be carried in a good pace. But I don't think, that something like that can happen. Spare time of a few volunteers is not enough.

**Comment by SethFalco** [comment]
As a regular user myself, I would be happy to help maintain svgo if you'd have me. I have availability to at least triage/resolve issues, fix reported bugs, improve documentation etc.

I've opened issues before, and have recently opened some pull requests. I understand it may not be enough yet, but just wanted to have my name down as interested already.

* https://github.com/svg/svgo/pull/1783
* https://github.com/svg/svgo/pull/1784
* https://github.com/svg/svgo/pull/1785

**Comment by deepsweet** [comment]
@SethFalco hi! You should be able to manage issues and PRs now. Please let me know when you're ready to publish a new version to NPM, I'll provide access token.

**Comment by SethFalco** [comment]
@deepsweet Hey! Sorry to pester, I just want to verify your intent here!

> let me know when you're ready to publish a new version to NPM

You've granted triage access, so I can manage issues and PRs. But only actions like approving or requesting changes, or cleaning up invalid issues. It does not allow running workflows on pull requests or write access, which would be normal to justify prompting for a new version.

Can you clarify if you intended to grant triage access or write access?

**Comment by deepsweet** [comment]
@SethFalco fixed, changed access to write.

**Comment by TrySound** [comment]
You can communicate with us in discord

**Comment by SethFalco** [comment]
> You can communicate with us in discord

I actually stray away of Discord and deleted my account some time ago!
I should've clarified that in my initial post, as I did notice the Discord link in the README. 😅

Let me know if that's a problem. Meanwhile, I'm always available on [Matrix](https://matrix.to/#/@sethi:one.ems.host), [Mastodon](https://fosstodon.org/@sethi), and [email](mailto:seth@falco.fun). ^-^'
Would also be more than happy to participate if it's ever [bridged](https://t2bot.io/discord/) over!

**Comment by danielbayley** [comment]
@deepsweet I’m already _way_ overextend currently 😅 But once the dust settles, I have much to contribute to this often used project…

**Comment by MathiasReker** [comment]
Hello

’d be happy to lend a helping hand with SVGO. I’m the author of the PHP SVG Optimizer (https://github.com/MathiasReker/php-svg-optimizer), and I’ve had a lot of experience working with SVG optimizations and related tools.

**Comment by KrasnayaPloshchad** [comment]
@waywardmonkeys Would you like to pay interest to here?

**Comment by KrasnayaPloshchad** [comment]
@LaurenzV @DJMcNab @JosefKuchar Would you like to pay interest to here?

**Comment by SethFalco** [comment]
Hey @deepsweet!

Would it be alright to invite @KTibow to become a maintainer of SVGO? I've had some trouble offering time to the project as of late (as my history demonstrates). Kendell has made many great contributions, not only in PRs, but responding to and fixing issues reported by users too. He almost certainly would've made more contributions if my inactivity didn't bottleneck them either. 😅

I'll still be around, and hope to offer more time to the project in future. But meanwhile, having another pair of hands will do the project a lot of good! 💪(•ᴗ•💪)

I've spoken with Kendall in private already just to gauge interest, and he's confirmed he'd be up for it! 👍

**Comment by deepsweet** [comment]
@SethFalco hey! The repo write access is ready, ping me with NPM username please.

**Comment by SethFalco** [comment]
Thank you! ❤️ 

We should be able to hold off on the npm access for now actually! One of the changes I made is that a release is made to npm automatically upon creating a release on GitHub. 

Reference: https://github.com/svg/svgo/pull/2181

So it should be enough for new maintainers to have the ability to create new releases/tags on GitHub! 🤞

**Comment by GreLI** [comment]
Still it would be good to have an access to npm in case of emergencies, if something breaks, etc.
